container/broker/broker.go

```diff
@@ -261,3 +265,33 @@ func proxyConfigurationFromContainerCfg(cfg params.ContainerConfig) instancecfg.
 		SnapStoreProxyURL:   cfg.SnapStoreProxyURL,
 	}
 }
+
+// ovsManagedBridges returns a filtered version of ifaceList that only contains
+// bridge interfaces managed by openvswitch.
+func ovsManagedBridges(ifaceList corenetwork.InterfaceInfos) (corenetwork.InterfaceInfos, error) {
+	if _, err := exec.LookPath("ovs-vsctl"); err != nil {
+		// ovs tools not installed; nothing to do
+		if execErr, isExecErr := err.(*exec.Error); isExecErr && execErr.Unwrap() == exec.ErrNotFound {
+			return nil, nil
+		}
+
+		return nil, errors.Annotate(err, "ovsManagedBridges: looking for ovs-vsctl")
+	}
+
+	// Query list of ovs-managed device names
+	res, err := getCommandOutput(exec.Command("ovs-vsctl", "list-br"))
+	if err != nil {
+		return nil, errors.Annotate(err, "querying ovs-managed bridges via ovs-vsctl")
+	}
+
+	ovsBridges := set.NewStrings()
+	for _, iface := range strings.Split(string(res), "\n") {
+		if iface = strings.TrimSpace(iface); iface != "" {
+			ovsBridges.Add(iface)
+		}
+	}
+
+	return ifaceList.Filter(func(iface corenetwork.InterfaceInfo) bool {
+		return ovsBridges.Contains(iface.InterfaceName)
```

Comment:
Does the command list tags added to the bridge as bridge devices?
Such as:
- `ovsbr1` (the main OVS bridge).
- `ovsbr1.100` (device connected to VLAN with tag 100).
- `ovsbr1.666` (device connected to VLAN with tag 666).
- etc.

`InterfaceInfo` has a method `ActualInterfaceName` that does the name/tag joining.

Just a thought; it should work either way.

Reply:
Yeah, saw that method. However, the command (at least this is what happens in my tests) just lists the names that  get added via `ovs-vsctl add-br X` so I am only using the device names for now. If it turns out that it doesn't work that way, I will revisit this.